app/controllers/project_user_roles_controller.rb

```diff
@@ -19,7 +26,14 @@ def find_project
     authorize Project.find(params[:project_id]), :edit?, policy_class: ProjectPolicy
   end
 
+  def find_user(project)
+    AuthInfo
+      .joins(:user)
+      .where(users: { auth_provider: project.integration_type }) # TODO: remove when we will merge users with different providers
+      .find_by("data->>'email' = ?", params[:project_user_role][:email])&.user
```

Comment:
nit: Use `User.join(:auth_info).where(...)` instead of `AuthInfo.join(:user).where(...) ... &.user`